File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailModel.java
package com.ed.redditapp.ui.post_detail;

import com.ed.redditapp.lib.api.RedditApi;

public class PostDetailModel {
    private final PostDetailCommentsAsyncTask commentsAsyncTask;

    public PostDetailModel(RedditApi redditApi) {
        commentsAsyncTask = new PostDetailCommentsAsyncTask(redditApi);
    }

    public void getPostComments(String postUrl, PostDetailCommentsAsyncTask.Callback callback) {
        commentsAsyncTask.run(postUrl, callback);
    }
}

File: app/src/main/java/com/ed/redditapp/ui/post_detail/PostDetailModel.kt
package com.ed.redditapp.ui.post_detail

import com.ed.redditapp.lib.api.RedditApi

class PostDetailModel(redditApi: RedditApi) {
    private val commentsAsyncTask = PostDetailCommentsAsyncTask(redditApi)

    fun getPostComments(postUrl: String, callback: PostDetailCommentsCallback) {
        commentsAsyncTask.run(postUrl, callback)
    }
}